### PR TITLE
Add table aws_pipes_pipe closes #1478

### LINKED ISSUE #1478: Add table aws_pipes_pipe
**References**
[Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.](https://github.com/aws/aws-sdk-go-v2/tree/main/service/pipes)

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>

```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select
  name,
  arn,
  current_state,
  creation_time,
  role_arn
from
  aws_pipes_pipe;
+--------+--------------------------------------------------+---------------+---------------------------+-------------------------------------------------------------------------------------+
| name   | arn                                              | current_state | creation_time             | role_arn                                                                            |
+--------+--------------------------------------------------+---------------+---------------------------+-------------------------------------------------------------------------------------+
| test23 | arn:aws:pipes:us-east-1:632902152528:pipe/test23 | RUNNING       | 2022-12-06T20:10:40+05:30 | arn:aws:iam::632902152528:role/service-role/Amazon_EventBridge_Pipe_test23_33df0164 |
+--------+--------------------------------------------------+---------------+---------------------------+-------------------------------------------------------------------------------------+

> select
  name,
  arn,
  description,
  creation_time,
  current_state,
  desired_state
from
  aws_pipes_pipe
where
  desired_state = current_state;
+--------+--------------------------------------------------+-------------+---------------------------+---------------+---------------+
| name   | arn                                              | description | creation_time             | current_state | desired_state |
+--------+--------------------------------------------------+-------------+---------------------------+---------------+---------------+
| test23 | arn:aws:pipes:us-east-1:632902152528:pipe/test23 | <null>      | 2022-12-06T20:10:40+05:30 | RUNNING       | RUNNING       |
+--------+--------------------------------------------------+-------------+---------------------------+---------------+---------------+

> select
  name,
  creation_time,
  current_state,
  desired_state,
  enrichment,
  target
from
  aws_pipes_pipe
where
  creation_time >= now() - interval '30' day;
+--------+---------------------------+---------------+---------------+-----------------------------------------------------------------------------------------------------+--------------------------------
| name   | creation_time             | current_state | desired_state | enrichment                                                                                          | target                         
+--------+---------------------------+---------------+---------------+-----------------------------------------------------------------------------------------------------+--------------------------------
| test23 | 2022-12-06T20:10:40+05:30 | RUNNING       | RUNNING       | arn:aws:lambda:us-east-1:632902152528:function:SecurityLake_Glue_Partition_Updater_Lambda_us-east-1 | arn:aws:events:us-east-1:632902
+--------+---------------------------+---------------+---------------+-----------------------------------------------------------------------------------------------------+--------------------------------
```
</details>

### CODE REVIEW

File: aws/table_aws_pipes_pipe.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+			},
+			{
+				Name:        "target_prefix",
+				Description: "The prefix matching the pipe target.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromQual("target_prefix"),
+			},
+			{
+				Name:        "enrichment_parameters",
+				Description: "The ARN of the target resource.",
```
Comment (misraved) [inline aws/table_aws_pipes_pipe.go:114]: Is the description correct?

File: aws/table_aws_pipes_pipe.go [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+	}
+}
+
+//// LIST FUNCTION
+
+func listAwsPipes(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Get client
+	svc, err := PipesClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_pipes_pipe.listAwsPipes", "get_client_error", err)
```
Comment (misraved) [inline aws/table_aws_pipes_pipe.go:154]:
```suggestion
		plugin.Logger(ctx).Error("aws_pipes_pipe.listAwsPipes", "client_error", err)
```

File: aws/table_aws_pipes_pipe.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+	}
+
+	if d.KeyColumnQualString("current_state") != "" {
+		params.CurrentState = types.PipeState(d.KeyColumnQualString("current_state"))
+	}
+	if d.KeyColumnQualString("desired_state") != "" {
+		params.DesiredState = types.RequestedPipeState(d.KeyColumnQualString("desired_state"))
+	}
+	if d.KeyColumnQualString("source_prefix") != "" {
+		params.SourcePrefix = aws.String(d.KeyColumnQualString("desired_state"))
```
Comment (misraved) [inline aws/table_aws_pipes_pipe.go:184]:
```suggestion
		params.SourcePrefix = aws.String(d.KeyColumnQualString("source_prefix"))
```

File: aws/table_aws_pipes_pipe.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+}
+
+//// HYDRATE FUNCTIONS
+
+func getAwsPipe(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	// Create Session
+	svc, err := PipesClient(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_pipes_pipe.getAwsPipe", "get_client_error", err)
```
Comment (misraved) [inline aws/table_aws_pipes_pipe.go:224]:
```suggestion
		plugin.Logger(ctx).Error("aws_pipes_pipe.getAwsPipe", "client_error", err)
```

File: docs/tables/aws_pipes_pipe.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
+# Table: aws_pipes_pipe
+
+Amazon EventBridge Pipes connects sources to targets. It reduces the need for specialized knowledge and integration code when developing event driven architectures, fostering consistency across your company’s applications. To set up a pipe, you choose the source, add optional filtering, define optional enrichment, and choose the target for the event data.
```
Comment (misraved) [inline docs/tables/aws_pipes_pipe.md:3]:
```suggestion
Amazon EventBridge Pipes connect sources to targets. It reduces the need for specialized knowledge and integration code when developing event-driven architectures, fostering consistency across your company’s applications. To set up a pipe, you choose the source, add optional filtering, define optional enrichment, and choose the target for the event data.
```

File: docs/tables/aws_pipes_pipe.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  desired_state <> current_state;
+```
+
+### Get the target parameters information for each pipe
+
+```sql
+select
+  name,
+  target_parameters ->> 'BatchJobParameters' as batch_job_parameters,
+  target_parameters ->> 'CloudWatchLogsParameters' as cloud_watch_logs_parameters,
```
Comment (misraved) [inline docs/tables/aws_pipes_pipe.md:42]:
```suggestion
  target_parameters ->> 'CloudWatchLogsParameters' as cloudwatch_logs_parameters,
```

File: docs/tables/aws_pipes_pipe.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  target_parameters ->> 'EcsTaskParameters' as ecs_task_parameters,
+  target_parameters ->> 'EventBridgeEventBusParameters' as event_bridge_event_bus_parameters,
```
Comment (misraved) [inline docs/tables/aws_pipes_pipe.md:44]:
```suggestion
  target_parameters ->> 'EventBridgeEventBusParameters' as eventbridge_event_bus_parameters,
```

File: docs/tables/aws_pipes_pipe.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+  target_parameters ->> 'LambdaFunctionParameters' as lambda_function_parameters,
+  target_parameters ->> 'RedshiftDataParameters' as redshift_data_parameters,
+  target_parameters ->> 'SageMakerPipelineParameters' as sage_maker_pipeline_parameters,
+  target_parameters ->> 'SqsQueueParameters' as sqs_queue_parameters,
+  target_parameters ->> 'StepFunctionStateMachineParameters' as step_function_state_machine_parameters
+from
+  aws_pipes_pipe;
+```
+
+## Get enrichment parameters details for each pipe
```
Comment (misraved) [inline docs/tables/aws_pipes_pipe.md:57]:
```suggestion
## Get enrichment parameter details for each pipe
```

Comment (misraved) [inline docs/tables/aws_pipes_pipe.md:57]:
```suggestion
### Get enrichment parameter details for each pipe
```

File: docs/tables/aws_pipes_pipe.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+```sql
+select
+  name,
+  enrichment_parameters ->> 'HttpParameters' as http_parameters,
+  enrichment_parameters ->> 'InputTemplate' as input_template
+from
+  aws_pipes_pipe;
+```
+
+## List pipes created within last 30 days
```
Comment (misraved) [inline docs/tables/aws_pipes_pipe.md:68]:
```suggestion
### List pipes created within the last 30 days
```

File: docs/tables/aws_pipes_pipe.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+  desired_state,
+  enrichment,
+  target
+from
+  aws_pipes_pipe
+where
+  creation_time >= now() - interval '30' day;
+```
+
+## Get role details for pipes
```
Comment (misraved) [inline docs/tables/aws_pipes_pipe.md:84]:
```suggestion
### Get IAM role details for pipes
```

**Review by misraved — COMMENTED** [review]
Please take a look at the review comments. Thanks!!!

**Comment by misraved** [comment]
@ParthaI I see a `NextToken` parameter being returned by the API, shouldn't we be adding the pagination to it?
https://docs.aws.amazon.com/eventbridge/latest/pipes-reference/API_ListPipes.html

**Comment by ParthaI** [comment]
> @ParthaI I see a `NextToken` parameter being returned by the API, shouldn't we be adding the pagination to it? https://docs.aws.amazon.com/eventbridge/latest/pipes-reference/API_ListPipes.html

@misraved We are already performing pagination in this PR by passing `NextToken` to input param.
https://github.com/turbot/steampipe-plugin-aws/pull/1487/files#diff-302694c891837b11a56f3bad73b359c9f5f11ec30df73b43552019302b0384cbR206-R209

**Review by misraved — APPROVED** [review]

**Comment by misraved** [comment]
Terraform table reference - https://github.com/hashicorp/terraform-provider-aws/issues/28153#issuecomment-1341064256

**Pull request merged**